packages/core/components/badge/index.js

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import { space, color, fontSize, borderRadius } from "styled-system";
+
+const StyledBadge = styled.span`
+  ${space}
+  ${color}
+  ${fontSize}
+  ${borderRadius}
+  display: inline-block;
+  line-height: 1;
+  text-align: center;
+  text-decoration: none;
+  text-transform: uppercase;
+`;
+
+const Badge = props => {
+  return (
+    <StyledBadge {...props}>{props.children}</StyledBadge>
+  );
+};
+
+Badge.propTypes = {
+  ...space.PropTypes,
+  ...color.PropTypes,
+  ...fontSize.PropTypes,
+  ...borderRadius.PropTypes
+};
+
+Badge.defaultProps = {
+  mx: 1,
+  my: "auto",
+  px: "0.4em",
+  py: "0.24em",
```

Comment:
I just ran into this issue myself, but when you define defaultProps for `mx`, `my`, `px`, `py` setting overrides with `m` or `p` no longer works correctly. This happened with the Icon component. I think it's a bug with styled-system.

The work around is to instead set defaults for `m` and `p`:
```
m: [1, "auto"],
p: ["0.4em", "0.24em"],
```

Reply:
Doh. I ran into this on a previous PR and solved it similarly, then somehow forgot all about it and fell into the same trap here.